contractcourt/channel_arbitrator.go

```diff
@@ -244,11 +260,44 @@ func (c *ChannelArbitrator) Start() error {
 		return err
 	}
 
+	// Check the database for the close status of this channel.
+	isClosing, closeType, err := c.cfg.IsPendingClose()
+	if err != nil {
+		c.cfg.BlockEpochs.Cancel()
+		return err
+	}
+
+	// If the channel has been marked pending close in the database, and we
+	// haven't transitioned the state machine to StateContractClosed (or a
+	// suceeding state), then a state transition most likely failed. We'll
+	// try to recover from this by manually advancing the state by setting
+	// the corresponding close trigger.
+	trigger := chainTrigger
+	if isClosing {
+		switch c.state {
+		case StateDefault:
+			fallthrough
+		case StateBroadcastCommit:
+			fallthrough
+		case StateCommitmentBroadcasted:
+			switch closeType {
+			case channeldb.LocalForceClose:
+				trigger = localCloseTrigger
+			case channeldb.RemoteForceClose:
+				trigger = remoteCloseTrigger
+			}
+
+			log.Warnf("ChannelArbitrator(%v): detected stalled "+
+				"state=%v for closed channel, using "+
+				"trigger=%v", c.cfg.ChanPoint, c.state, trigger)
+		}
+	}
+
 	// We'll now attempt to advance our state forward based on the current
 	// on-chain state, and our set of active contracts.
 	startingState := c.state
 	nextState, _, err := c.advanceState(
-		uint32(bestHeight), chainTrigger, nil,
+		uint32(bestHeight), trigger, nil,
```

Comment:
do we always want to pass in `bestHeight` if it's not a `chainTrigger`? may need to pass in spend height?